### ISSUE TITLE
GRAPHICS: Deadlock with anti-aliasing enabled

### ISSUE THREAD

**Issue by DrMcCoy** [xoreos/xoreos#506]
When enabling antialiasing (with, for example, the command line options `--fsaa 2`), xoreos currently deadlocks in Graphics::GraphicsManager::setFSAA().

Backtrace of the main xoreos thread:

```
#0  0x00007ffff718111c in futex_wait_cancelable (private=<optimized out>, expected=0,·
    futex_word=0x60f0000073c0) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  __pthread_cond_wait_common (abstime=0x0, mutex=0x6060000189f0, cond=0x60f000007398)
    at pthread_cond_wait.c:502
#2  __pthread_cond_wait (cond=0x60f000007398, mutex=0x6060000189f0) at pthread_cond_wait.c:655
#3  0x00007ffff4af7dfc in __gthread_cond_wait (__mutex=<optimized out>, __cond=<optimized out>)
    at /var/tmp/portage/sys-devel/gcc-8.3.0/work/build/x86_64-pc-linux-gnu/libstdc++-v3/include/x86_64-pc-linux-gnu/bits/gthr-default.h:864
#4  std::condition_variable::wait (this=<optimized out>, __lock=...)
    at /var/tmp/portage/sys-devel/gcc-8.3.0/work/gcc-8.3.0/libstdc++-v3/src/c++11/condition_variable.cc:53
#5  0x000055555af5ad8b in std::_V2::condition_variable_any::wait<std::unique_lock<std::recursive_mutex> > (
    __lock=..., this=<optimized out>)
    at /usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8/bits/std_mutex.h:238
#6  Events::RequestManager::waitReply (this=<optimized out>, request=...)
    at /home/drmccoy/projects/xoreos/xoreos/src/events/requests.cpp:90
#7  0x000055555af5d484 in Events::RequestManager::dispatchAndWait (Python Exception <class 'ValueError'> Cannot find type Events::RequestID::_Node:·
request=, this=0x60d000003b90)
    at /home/drmccoy/projects/xoreos/xoreos/src/events/requests.cpp:115
#8  Events::RequestManager::sync (this=0x60d000003b90)
    at /home/drmccoy/projects/xoreos/xoreos/src/events/requests.cpp:126
#9  0x000055555b47d7ff in Graphics::GraphicsManager::setFSAA (this=0x619000021780, level=2)
    at /home/drmccoy/projects/xoreos/xoreos/src/graphics/graphics.cpp:236
#10 0x000055555b4828a0 in Graphics::GraphicsManager::init (this=this@entry=0x619000021780)
    at /home/drmccoy/projects/xoreos/xoreos/src/common/singleton.h:91
#11 0x0000555559ca02da in init () at /home/drmccoy/projects/xoreos/xoreos/src/common/singleton.h:91
#12 main (argc=<optimized out>, argv=<optimized out>)
    at /home/drmccoy/projects/xoreos/xoreos/src/xoreos.cpp:189
```

and the SDL timer thread:

```
#0  0x00007ffff71841b6 in futex_abstimed_wait_cancelable (private=0, abstime=0x0, expected=0,·
    futex_word=0x6030000062b0) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
205→in ../sysdeps/unix/sysv/linux/futex-internal.h
(gdb) bt
#0  0x00007ffff71841b6 in futex_abstimed_wait_cancelable (private=0, abstime=0x0, expected=0,·
    futex_word=0x6030000062b0) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
#1  do_futex_wait (sem=sem@entry=0x6030000062b0, abstime=0x0) at sem_waitcommon.c:111
#2  0x00007ffff71842c8 in __new_sem_wait_slow (sem=0x6030000062b0, abstime=0x0) at sem_waitcommon.c:181
#3  0x00007ffff6b70902 in SDL_SemWait_REAL (sem=0x6030000062b0)
    at /var/tmp/portage/media-libs/libsdl2-2.0.9/work/SDL2-2.0.9/src/thread/pthread/SDL_syssem.c:95
#4  0x00007ffff6b70a1d in SDL_SemWaitTimeout_REAL (sem=0x6030000062b0, timeout=4294967295)
    at /var/tmp/portage/media-libs/libsdl2-2.0.9/work/SDL2-2.0.9/src/thread/pthread/SDL_syssem.c:126
#5  0x00007ffff6b096fa in SDL_TimerThread (_data=_data@entry=0x7ffff6bc1d20 <SDL_timer_data>)
    at /var/tmp/portage/media-libs/libsdl2-2.0.9/work/SDL2-2.0.9/src/timer/SDL_timer.c:201
#6  0x00007ffff6b0918c in SDL_RunThread (data=0x6030000062e0)
    at /var/tmp/portage/media-libs/libsdl2-2.0.9/work/SDL2-2.0.9/src/thread/SDL_thread.c:283
#7  0x00007ffff6b70479 in RunThread (data=<optimized out>)
    at /var/tmp/portage/media-libs/libsdl2-2.0.9/work/SDL2-2.0.9/src/thread/pthread/SDL_systhread.c:78
#8  0x00007ffff717a46a in start_thread (arg=0x7fffeeb9b700) at pthread_create.c:486
#9  0x00007ffff3c99f3f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by Supermanu** [comment]
This regression is introduced by 0cfd5b7d70839da5a5123b9852b326a86f644e43.

**Comment by DrMcCoy** [comment]
(Btw, I do consider this a blocker for a new release. I haven't yet had time to look at it, so if anyone else wants to take this, feel free.)

**Comment by seedhartha** [comment]
xoreos with antialiasing works fine on Windows, so it could be a Unix specific regression.

**Comment by DrMcCoy** [comment]
Damn, that doesn't make it easier to pinpoint :P

**Comment by seedhartha** [comment]
Fixed by #539.
